### PR TITLE
gha: enable autogenerated policy testing on SEV and SEV-SNP

### PR DESCRIPTION
**Pull request by Redent0r**
Enable autogenerated policy testing on SEV and SEV-SNP

### CODE REVIEW

**Comment by Redent0r** [comment]
Needs https://github.com/kata-containers/kata-containers/pull/9844

File: tests/integration/kubernetes/k8s-policy-pvc.bats [hunk 1]
```diff
@@ -9,6 +9,7 @@ load "${BATS_TEST_DIRNAME}/../../common.bash"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
 
 setup() {
+	( [ "${KATA_HYPERVISOR}" == "qemu-sev" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ] ) && skip "https://github.com/kata-containers/kata-containers/issues/9846"
```
Comment (danmihai1) [inline tests/integration/kubernetes/k8s-policy-pvc.bats:12]: Let's move the new skip after the older skip below. No need to point to the PVC issue if auto_generate was disabled.

**Review by danmihai1 — APPROVED** [review]

**Review by AdithyaKrishnan — APPROVED** [review]
LGTM

File: tests/integration/kubernetes/tests_common.sh [hunk 2]
```diff
@@ -130,6 +130,39 @@ auto_generate_policy_enabled() {
 	[ "${AUTO_GENERATE_POLICY}" == "yes" ]
 }
 
+# adapt common policy settings for qemu-snp
+adapt_common_policy_settings_for_snp() {
```
Comment (danmihai1) [inline tests/integration/kubernetes/tests_common.sh:134]: Is adapt_common_policy_settings_for_snp identical to adapt_common_policy_settings_for_sev?

If yes, let's keep just adapt_common_policy_settings_for_sev and call it on both KATA_HYPERVISOR branches.

**Review by danmihai1 — CHANGES_REQUESTED** [review]

File: tests/integration/kubernetes/tests_common.sh (hunk 2 above)
Comment (Redent0r) [reply tests/integration/kubernetes/tests_common.sh]: Updated. Thanks!

File: tests/integration/kubernetes/k8s-policy-pvc.bats (hunk 1 above)
Comment (Redent0r) [reply tests/integration/kubernetes/k8s-policy-pvc.bats]: Updated

**Review by danmihai1 — APPROVED** [review]

File: tests/integration/kubernetes/tests_common.sh [hunk 3]
```diff
@@ -147,6 +147,10 @@ adapt_common_policy_settings() {
 	case "${KATA_HYPERVISOR}" in
   		"qemu-sev")
 			adapt_common_policy_settings_for_sev "${settings_dir}"
+			;;
+		"qemu-snp")
```
Comment (wainersm) [inline tests/integration/kubernetes/tests_common.sh:151]: You can merge it in the previous qemu-sev block, like:

```
qemu-sev|qemu-snp)
   adapt_common_policy_settings_for_sev "${settings_dir}"
...
...
```

**Comment by wainersm** [comment]
please see the comments that I left in https://github.com/kata-containers/kata-containers/pull/9812

File: tests/integration/kubernetes/tests_common.sh (hunk 3 above)
Comment (Redent0r) [reply tests/integration/kubernetes/tests_common.sh]: Updated. Thanks @wainersm  !

**Review by ryansavino — APPROVED** [review]

**Pull request merged**